platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/internal/CheckstyleInvoker.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality.internal;
+
+import com.google.common.collect.ImmutableMap;
+import groovy.namespace.QName;
+import groovy.util.Node;
+import groovy.util.NodeList;
+import groovy.xml.XmlParser;
+import org.apache.commons.io.IOUtils;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.internal.exceptions.MarkedVerificationException;
+import org.gradle.api.internal.project.antbuilder.AntBuilderDelegate;
+import org.gradle.api.plugins.quality.Checkstyle;
+import org.gradle.api.provider.Property;
+import org.gradle.internal.logging.ConsoleRenderer;
+import org.gradle.util.GradleVersion;
+import org.gradle.util.internal.GFileUtils;
+import org.gradle.util.internal.VersionNumber;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.gradle.api.plugins.quality.internal.AntInvokeUtils.invoke;
+
+class CheckstyleInvoker implements Action<AntBuilderDelegate> {
```

Comment:
💭 This is just rewritten from Groovy to Java so we can use annotation processor for code-quality project